### PR TITLE
Add @protosaurus/{generator,go,home,wkt,example}

### PR DESCRIPTION
**Pull request by dio**
This adds several packages for generating *.json from *.proto:

1. `@protosaurus/generator` essentially is `buf generate` as a library.
2. `@protosaurus/go` makes sure we have Go installed in the current machine (so we can install buf and protoc-gen-doc without worrying about system os and arch).
2. `@protosaurus/home` sets several required directories for protosaurus to run. 
3. `@protosaurus/wkt` bundles generated WKT, this is versioned the same as protoc that is used for generating the *.json (currently 3.9.2).
4. `@protosaurus/example` acts as a demo for generating *.json from *.proto, defined by this project buf.work.yaml (that points to `testdata/buf.yaml`) by using the `@protosaurus/generator`.

A notable change: as you can see in the main Makefile (see `gen:` target), we now _can_ take advantage of `@protosaurus/example`!

```
gen: $(BUF_V1_MODULE_DATA) $(yarn) ## Generate files from proto files
    ...
	@$(yarn) workspace @protosaurus/example generate
    ...
```

Signed-off-by: Dhi Aurrahman <dio@rockybars.com>

### CODE REVIEW

**Comment by dio** [comment]
The quality of the code is really still a sketch, but this paved the way to e2e.

**Comment by dio** [comment]
Most of the files in this PR are WKT .json files!

File: package.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "name": "protosaurus",
+  "version": "1.0.0",
+  "private": true,
+  "workspaces":[
+    "website/*",
+    "packages/*"
+  ],
+  "engines": {
+    "node": "16.13.2",
+    "yarn": "1.22.17"
```
Comment (imballinst) [inline package.json:11]: I was just about to comment on using a "less strict version", but apparently `engines` field is not strict, TIL! https://docs.npmjs.com/cli/v6/configuring-npm/package-json#engines

**Review by imballinst — APPROVED** [review]
LGTM! Thanks for working on this.

**Comment by dio** [comment]
Thanks, @imballinst!

**Pull request merged**